psutil/_psosx.py

```diff
@@ -212,6 +213,91 @@ def disk_partitions(all=False):
 # =====================================================================
 
 
+def get_smc_temperatures(smc_keys, iter_num, iteratable):
+    """ Receives a dictionary of SMC keys, and a number of possible keys.
+    If keys can be iterated, replaces the expected %i in the key.
+    Returns a list of all smc_get_temperature results higher than 0.
+    References:
+
+    https://github.com/Chris911/iStats/blob/
+        09b159f85a9481b59f347a37259f6d272f65cc05/lib/iStats/smc.rb
+
+    http://web.archive.org/web/20140714090133/http://www.parhelia.ch:80/
+        blog/statics/k3_keys.html
+    """
+    ret = list()
+    if iteratable:
+        for idx in range(iter_num):
+            for key, label in smc_keys.items():
+                result = cext.smc_get_temperature(key % idx)
+                result = round(result, 1)
+                if result <= 0:
+                    continue
+                ret.append((label % idx, result, None, None))
+    else:
+        for key, label in smc_keys.items():
+            result = cext.smc_get_temperature(key)
+            result = round(result, 1)
+            if result <= 0:
+                continue
+            ret.append((label, result, None, None))
+
+    return ret
+
+
+def sensors_temperatures():
+    """Return CPU temperatures including key name and temperature.  """
+    ret = collections.defaultdict(list)
+    # Handle CPU cores
+    num_cpus = cext.cpu_count_phys()
+    core_temps = {
+        "TC%iC": "Core %i",
+    }
+    if num_cpus:
+        results = get_smc_temperatures(core_temps, num_cpus + 1, True)
+        ret["CPU"] = ret["CPU"] + results
+
+    # Handle other CPU sensors
+    cpu_temps = {
+        "TC%iF": "CPU TC%iF",
+        "TC%iG": "CPU TC%iG",
+        "TC%iP": "CPU %i Proximity",
+    }
+    num_packages = cext.cpu_count_packages()
+    if num_packages:
+        results = get_smc_temperatures(cpu_temps, num_packages, True)
+        ret["CPU"] = ret["CPU"] + results
+
+    # Handle Battery Temperature
+    battery_temps = {
+        "TB%iT": "Battery %i",
+    }
+    # 2 is the highest number of batteries recoreded in iStats
+    results = get_smc_temperatures(battery_temps, 3, True)
+    ret["Battery"] = ret["Battery"] + results
+
+    # Handle HDD temperatures
+    hdd_temps = {
+        "TH%iP": "HDD %i Proximity",
+        "TH%ia": "HDD %i Unknown",
```

Comment:
It's probably better to remove the "Unknown" part. Also I would add a comment like:

```python
# XXX: HWSensors is able to determine drive name
```